### ISSUE TITLE
Bias in z dimension with 3d bbox prediction

### ISSUE THREAD

**Issue by bmankirLinker** [traveller59/second.pytorch#236]
I'm getting accurate bounding boxes in terms of all values but the ` z ` translation of 3d bounding box when using my own data, does anyone know does it require to make any preprocessing based on the configuration of the lidar itself (such as height) or normalization etc.

**Comment by micjabbour** [comment]
Are you using one of the models pre-trained on the KITTI dataset? or are you training on your own data?

**Comment by bmankirLinker** [comment]
@micjabbour I'm using the KITTI pre-trained model and inferring on my own dataset.

**Comment by micjabbour** [comment]
One quirk in the KITTI dataset is that the plane z=0 passes through the LIDAR sensor (the points on the ground usually have z=-1.73). Does that apply to your dataset too? or do you follow a different coordinate system?

If it is not, the safest way would be to apply a fixed translation on your dataset (to make the origin align with that of KITTI's), predict, then apply the inverse translation on the labels.

**Comment by bmankirLinker** [comment]
@micjabbour In my dataset z=0 passes through the LIDAR sensor, however, the LIDAR sensors height is a little lower around 1.5 meters thus the ground is around z=-1.5. 
But the bias is around 1 meter and also changes with the tweak of the PC_SCOPE parameter in the config.

**Comment by micjabbour** [comment]
I don't know much about SECOND, but in my opinion such a difference might cause the problem you are describing if you are using the PointPillars configuration files.

I would try to translate the point cloud scenes in your dataset to make them as close as possible to those of KITTI's and then try the prediction to see the results.

Hope this helps...
